src/AElf.OS.Network.Grpc/GrpcPeer.cs

```diff
@@ -250,17 +250,26 @@ private void HandleFailure(AggregateException exceptions, string errorMessage)
                     // Either we connected again or the state change wait timed out.
                     if (_channel.State == ChannelState.TransientFailure || _channel.State == ChannelState.Connecting)
                     {
-                        await StopAsync();
-                        DisconnectionEvent?.Invoke(this, EventArgs.Empty);
+                        await StopAndRemove();
                     }
                 });
             }
+            else if(exceptions.InnerException is RpcException rpcEx && rpcEx.StatusCode == StatusCode.Cancelled)
+            {
+                _ = StopAndRemove();
+            }
             else
             {
                 throw new NetworkException($"Failed request to {this}: {errorMessage}", exceptions);
             }
         }
 
+        private async Task StopAndRemove()
```

Comment:
StopAndRemoveAsync  

Async method should end with Async,

I hope it's my last comment of a name of function